jerry-main/main-unix-snapshot.c

```diff
@@ -0,0 +1,308 @@
+};
+
+/**
+ * Process 'merge' command.
+ *
+ * @return error code (0 - no error)
+ */
+static int
+process_merge (cli_state_t *cli_state_p, /**< cli state */
+               int argc, /**< number of arguments */
+               char *prog_name_p) /**< program name */
+{
+  jerry_init (JERRY_INIT_EMPTY);
+
+  uint8_t *input_pos_p = input_buffer;
+
+  cli_change_opts (cli_state_p, merge_opts);
+
+  const char *output_file_name_p = "merged.snapshot";
+  const uint32_t *merge_buffers[argc];
+  size_t merge_buffer_sizes[argc];
+  uint32_t number_of_files = 0;
+
+  for (int id = cli_consume_option (cli_state_p); id != CLI_OPT_END; id = cli_consume_option (cli_state_p))
+  {
+    switch (id)
+    {
+      case OPT_MERGE_HELP:
+      {
+        cli_help (prog_name_p, "merge", merge_opts);
+        return JERRY_STANDALONE_EXIT_CODE_OK;
+      }
+      case OPT_MERGE_OUT:
+      {
+        output_file_name_p = cli_consume_string (cli_state_p);
+        break;
+      }
+      case CLI_OPT_DEFAULT:
+      {
+        const char *file_name_p = cli_consume_string (cli_state_p);
+
+        if (cli_state_p->error == NULL)
+        {
+          size_t size = read_file (input_pos_p, file_name_p);
+
+          if (size == 0)
+          {
+            return JERRY_STANDALONE_EXIT_CODE_FAIL;
+          }
+
+          merge_buffers[number_of_files] = (const uint32_t *) input_pos_p;
+          merge_buffer_sizes[number_of_files] = size;
+
+          number_of_files++;
+          uintptr_t mask = sizeof (uint32_t) - 1;
+          input_pos_p = (uint8_t *) ((((uintptr_t) input_pos_p) + size + mask) & ~mask);
+        }
+        break;
+      }
+      default:
+      {
+        cli_state_p->error = "Internal error";
+        break;
+      }
+    }
+  }
+
+  if (cli_state_p->error != NULL)
+  {
+    if (cli_state_p->arg != NULL)
+    {
+      jerry_port_log (JERRY_LOG_LEVEL_ERROR, "Error: %s %s\n", cli_state_p->error, cli_state_p->arg);
+    }
+    else
+    {
+      jerry_port_log (JERRY_LOG_LEVEL_ERROR, "Error: %s\n", cli_state_p->error);
+    }
+
+    return JERRY_STANDALONE_EXIT_CODE_FAIL;
+  }
+
+  if (number_of_files < 2)
+  {
+    printf ("At least two input files must be passed.\n");
+
+    return JERRY_STANDALONE_EXIT_CODE_FAIL;
+  }
+
+  const char *error_p;
+  size_t size = jerry_merge_snapshots (merge_buffers,
+                                       merge_buffer_sizes,
+                                       number_of_files,
+                                       output_buffer,
+                                       JERRY_BUFFER_SIZE,
+                                       &error_p);
+
+  if (size == 0)
+  {
+    jerry_port_log (JERRY_LOG_LEVEL_ERROR, "Error: %s\n", error_p);
+    return JERRY_STANDALONE_EXIT_CODE_OK;
+  }
+
+  FILE *file_p = fopen (output_file_name_p, "w");
+
+  if (file_p != NULL)
+  {
+    fwrite (output_buffer, 1u, size, file_p);
+    fclose (file_p);
+  }
+  else
+  {
+    printf ("Cannot open file: '%s'\n", output_file_name_p);
```

Comment:
Only an error message is printed but returns OK afterwards. Should return with FAIL code. Moreover, code is mixing `jerry_port_log` and `printf` calls (not only here). It would be good to use only one (`jerry_port_log` seems to be the better choice at first sight).